docs/concepts/how-ipfs-works.md

```diff
@@ -5,86 +5,171 @@ description: Learn how the InterPlanetary File System (IPFS) works and why it's
 
-You've discovered your content, and you've found the current location(s) of that content. Now, you need to connect to that content and get it (_exchange_). To request blocks from and send blocks to other peers, IPFS currently uses a module called [_Bitswap_](https://github.com/ipfs/specs/blob/master/BITSWAP.md). Bitswap allows you to connect to the peer or peers that have the content you want, send them your _wantlist_ (a list of all the blocks you're interested in), and have them send you the blocks you requested. Once those blocks arrive, you can verify them by hashing their content to get CIDs and compare them to the CIDs that you requested. These CIDs also allow you to deduplicate blocks if needed.
+- _<VueCustomTooltip label="An IPFS HTTP Gateway that can fetch data from the IPFS network using the HTTP GET method." underlined multiline is-small is-right>Read-only</VueCustomTooltip>_
+- _<VueCustomTooltip label="An IPFS HTTP Gateway that can write data to the IPFS network using the HTTP POST, PUT and DELETE methods." underlined multiline is-small is-right>Writeable</VueCustomTooltip>_
+- _<VueCustomTooltip label="An IPFS HTTP Gateway that can limit access to and from data on the IPFS network by acting as a reverse proxy." underlined multiline is-small is-right>Authentication</VueCustomTooltip>_
 
-There are [other content replication protocols under discussion](https://github.com/ipfs/camp/blob/master/DEEP_DIVES/24-replication-protocol.md) as well, the most developed of which is [_Graphsync_](https://github.com/ipld/ipld/blob/master/specs/transport/graphsync/index.md). There's also a proposal under discussion to [extend the Bitswap protocol](https://github.com/ipfs/go-bitswap/issues/186) to add functionality around requests and responses.
+Additionally, there are multiple types of gateway providers:
 
-## SHA file hashes won't match Content IDs
+- _<VueCustomTooltip label="An IPFS HTTP Gateway hosted as a local service i.e. at localhost:8080 on your local machine." underlined multiline is-small is-right>Local gateway</VueCustomTooltip>_
+- _<VueCustomTooltip label="An IPFS HTTP Gateway hosted within a private network that handles access to the IPFS network for multiple computers within the private network, and may be configured to limit access to requests from specific domains." underlined multiline is-small is-right>Private gateway</VueCustomTooltip>_
+- _<VueCustomTooltip label="An IPFS HTTP Gateway hosted and operated by a third-party, such as Protocol Labs." underlined multiline is-small is-right>Public gateway</VueCustomTooltip>_
 
-You may be used to verifying the integrity of a file by matching SHA hashes, but SHA hashes won't match CIDs. Because IPFS splits a file into blocks, each block has its own CID, including separate CIDs for any parent nodes.
+### GraphSync
```

Comment:
As mentioned above, I would remove Garphsync as it isn't broadly implemented in IPFS